a10_neutron_lbaas/v2/handler_listener.py

```diff
@@ -272,10 +272,13 @@ def _delete_listener(self, c, context, listener, protocol):
 
     def _delete(self, c, context, listener):
         # First, remove any existing cert bindings
-        self._remove_existing_bindings(c, context, listener)
+
+        protocol = listener.protocol if self._remove_existing_bindings(
```

Comment:
`listener.protocol` can't be right, it has the wrong type. It's an openstack protocol. The argument to `_delete_listener` is an acos protocol.

Try this:

```
protocol = openstack_mappings.vip_protocols(c, listener.protocol)

if self._remove_existing_bindings(c, context, listener):
    protocol = c.client.slb.virtual_server.vport.HTTPS
```